extensions/ruby/src/language_servers/solargraph.rs

```diff
@@ -0,0 +1,121 @@
+                        if completion.label.starts_with(":") {
+                            "string.special.symbol"
+                        } else {
+                            "keyword"
+                        }
+                    }
+                    CompletionKind::Variable => {
+                        if completion.label.starts_with("@") {
+                            "property"
+                        } else {
+                            return None;
+                        }
+                    }
+                    _ => return None,
+                };
+
+                let len = completion.label.len();
+                let name_span =
+                    CodeLabelSpan::literal(completion.label, Some(highlight_name.to_string()));
+
+                Some(CodeLabel {
+                    code: Default::default(),
+                    spans: if let Some(detail) = completion.detail {
+                        vec![
+                            name_span,
+                            CodeLabelSpan::literal(" ", None),
+                            CodeLabelSpan::literal(detail, None),
+                        ]
+                    } else {
+                        vec![name_span]
+                    },
+                    filter_range: (0..len).into(),
+                })
+            }
+            _ => None,
+        }
+    }
+
+    pub fn label_for_symbol(&self, symbol: Symbol) -> Option<CodeLabel> {
+        let name = &symbol.name;
+
+        return match symbol.kind {
+            SymbolKind::Method => {
+                let mut parts = name.split('#');
+                let container_name = parts.next()?;
+                let method_name = parts.next()?;
+
+                if parts.next().is_some() {
+                    return None;
+                }
+
+                let filter_range = 0..name.len();
+
+                let spans = vec![
+                    CodeLabelSpan::literal(container_name, Some("type".to_string())),
+                    CodeLabelSpan::literal("#", None),
+                    CodeLabelSpan::literal(method_name, Some("function.method".to_string())),
+                ];
+
+                Some(CodeLabel {
+                    code: name.to_string(),
+                    spans,
+                    filter_range: filter_range.into(),
+                })
+            }
+            SymbolKind::Class | SymbolKind::Module => {
+                let class = "class ";
+                let code = format!("{class}{name}");
+                let filter_range = 0..name.len();
+                let display_range = class.len()..class.len() + name.len();
+
+                Some(CodeLabel {
+                    code,
+                    spans: vec![CodeLabelSpan::code_range(display_range)],
+                    filter_range: filter_range.into(),
+                })
+            }
+            SymbolKind::Constant => {
+                let code = format!("{}", name.to_uppercase());
```

Comment:
In Ruby, `constants` start with an uppercase letter, but I didn't want to make the implementation here more complicated, so uppercasing the `constant` should be good too